rectools/model_selection/cross_validate.py

```diff
@@ -137,7 +204,35 @@ def cross_validate(  # pylint: disable=too-many-locals
         else:
             item_ids_to_recommend = None
 
+        if ref_models is not None:
+            ref_reco = _get_ref_reco(
+                ref_models=ref_models,
+                models=models,
+                test_users=test_users,
+                fold_dataset=fold_dataset,
+                k=k,
+                filter_viewed=filter_viewed,
+                internal_item_ids_to_recommend=item_ids_to_recommend,
+            )
+            models_ref_reco = {model_name: None if model_name in ref_models else ref_reco for model_name in models}
+
+            metrics_without_intersection = {
+                metric_name: metric
+                for metric_name, metric in metrics.items()
+                if not isinstance(metric, IntersectionMetric)
+            }
+            models_metrics = {
+                model_name: metrics_without_intersection if model_name in ref_models else metrics
+                for model_name in models
+            }
+        else:
+            models_ref_reco = {model_name: None for model_name in models}
+            models_metrics = {model_name: metrics for model_name in models}
+
         for model_name, model in models.items():
+            if ref_models is not None and model_name in ref_models and not validate_ref_models:
+                continue
+
             model.fit(fold_dataset)
```

Comment:
here we can check if we have this model recos in `ref_reco` already. If we do -> get reco from there and not fit for the second time